siem_integrations/README.md

```diff
@@ -26,10 +26,103 @@ In order to utilize splunk2kafka, a [running Kafka instance](https://kafka.apach
 2. Install export2kafka ([Instructions](https://github.com/rapidsai/clx-siem-integration/blob/master/splunk2kafka/export2kafka/README.md))
 
 
-## CLX Query
-
-Coming soon...
-
+## CLX Query Sample Dataset
+
+1. Download sample dataset.
+    
+    ```aidl
+    wget http://files.grouplens.org/datasets/movielens/ml-25m.zip
+    ```
+   
+## CLX_Query_Service
+
+CLX Query Service is a restful service that runs using the django platform, which takes requests from the splunk clx query app and executes blazingsql query and returns the json response.
+
+### Install and Run CLX_Query_Service
+
+1. Install dependencies if not available.
+
+    ```aidl
+    pip install django gunicorn djangorestframework supervisor
+    ``` 
+2. Update settings by adding current query service running hostname.
+
+    ```aidl
+    vi clx_query_service/settings.py
+    ``` 
+    ```aidl
+    ALLOWED_HOSTS = ["<hostname goes here...>"]
+    ``` 
+3. Update configuration file with downloaded dataset location.
+   
+    ```aidl
+    vi clx_query_service/conf/clx_blz_reader_conf.yaml
+    ```
+    
+4. Start query service using gunicorn.
+
+     ```aidl
+     bash /clx/siem_integrations/clx_query_service/bin/start_service.sh --help
+     ```
+    ``` 
+    Usage: /clx/siem_integrations/clx_query_service/bin/start_service.sh [POS]... [ARG]...
+    Example-1: bash /clx/siem_integrations/clx_query_service/bin/start_service.sh -p 8998 -w 2 -t 60
+    Example-2: bash /clx/siem_integrations/clx_query_service/bin/start_service.sh --port 8990 --workers 4 --timeout 10
+    
+    CLX Query Service Runner...
+    
+    Positional:
+      -p, --port          Port number
+      -w, --workers       Number of workers
+      -t, --timeout       Application time out
+    
+      -h, --help          Print this help
+    ```
+5. Run gunicorn wrapped clx query service application as daemon process using supervisor.
+
+   ```aidl
+   cp /clx/siem_integrations/clx_query_service/conf/clx_query_service.conf /etc/supervisor
+   ```
+   ```aidl
+   supervisord -c /etc/supervisor/clx_query_service.conf
+   ```
+   ```aidl
+   supervisorctl -c supervisor/clx_query_service.conf
+   ```
+   ```aidl
+   start clx_query_service
+   ```
+
+## CLX_Query
+
+CLX Query is a splunk application with the ability to perform customized queries on the clx python module, which can trigger internal workflows to retrieve the requested analytical report using blazingsql engine.
```

Comment:
This can likely be moved up to the very beginning (before the sample dataset part).